tests/models.py

```diff
@@ -3392,6 +3392,26 @@ def test_connection_from_uri_no_schema(self):
         self.assertEqual(connection.password, 'password with space')
         self.assertEqual(connection.port, 1234)
 
+    def test_connection_from_uri_without_authinfo(self):
+        uri = 'scheme://host:1234'
+        connection = Connection(uri=uri)
+        self.assertEqual(connection.conn_type, 'scheme')
+        self.assertEqual(connection.host, 'host')
+        self.assertEqual(connection.schema, '')
+        self.assertEqual(connection.login, None)
+        self.assertEqual(connection.password, None)
+        self.assertEqual(connection.port, 1234)
```

Comment:
This is an additional test not directly related to the issues, but I think it is worth adding it to prevent regression.